File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnBackgroundDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnBackgroundDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd2
            AppTheme.STANDARD -> R.color.cstd2
            AppTheme.BLUE -> R.color.cbd2
            AppTheme.GREEN -> R.color.cgd2
            AppTheme.RED -> R.color.crd2
            AppTheme.NOTHING -> R.color.cnod2
        }
    }
}

File: lib/layer2_core/dark/e_on_background_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnBackgroundDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd2;
      case 'standard': return cstd2;
      case 'blue': return cbd2;
      case 'green': return cgd2;
      case 'red': return crd2;
      case 'nothing': return cnod2;
      default: return cstd2;
    }
  }
}
